### DISCUSSION TITLE
How to switch the launch type to EC2 instead of Fargate

### DISCUSSION THREAD

**Discussion by DennisSSDev** [aws/copilot-cli#1885] [Q&A] [answered]
Hi! I was working on an application's backend using copilot and have been doing some cost calculations and realized that at some point in the future we would need to switch to EC2 + ECS instead of Fargate to be more cost effective. Are there any ways to easily define the launch type to EC2 in the manifest YAML file? Would appreciate any pointers 😄

**Comment by efekarakus** [comment, marked as answer]
Hi @DennisSSDev ! Unfortunately we don't have support yet for the EC2 launch type 🙇 We'll update this issue when we start working on it: https://github.com/aws/copilot-cli/issues/1094

**Reply by kohidave** [reply]
Yea! And not to sound too much like a Fargate marketer but there are a couple of things that I think are worth mentioning in terms of cost:

1. It's much easier and quicker to scale Fargate tasks up and down based on need - so while in terms of raw instance hour price, EC2 is cheaper than Fargate, it's easier and quicker to scale up and down based on demand
2. We're pretty close to introducing Fargate Spot support into Copilot - which is substantially cheaper than regular Fargate. This is especially useful for scaling scenarios and may be worth thinking about :)
3. It's worth considering the dev time spent managing, patching and monitoring raw EC2 instances. 

Anyway, those are just some things to consider - but as Efe mentioned, we're tracking EC2 support in #1094 

Thanks @DennisSSDev for the question!!
